### PR TITLE
[New Topic] Luau

### PR DESCRIPTION
**Pull request by KTom101**
### Description

Add a Luau topic page so new concepts can be created and existing functions can be correctly identified under Luau.

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/luau/luau.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+Title: 'Luau'
+Description: 'Luau is a scripting language derived from Lua. It was created for Roblox, an online game platform and game creation system. Luau is backwards compatible with Lua 5.1, meaning Lua 5.1 can be executed within the Luau environment.'
+Codecademy Hub Page: 'https://www.codecademy.com/catalog/language/luau'
+CatalogContent:
+  - 'paths/computer-science'
+---
+
+Luau is a scripting language derived from Lua. It was created for Roblox, an online game platform and game creation system. Luau is backwards compatible with Lua 5.1, meaning Lua 5.1 can be executed within the Luau environment. Luau is used by Roblox game developers to write game code and by Roblox engineers to write large parts of the user-facing application code and also portions of the Roblox Studio editor as plugins.
```
Comment (yangc95) [inline content/luau/luau.md:9]: Can we have a ## Luau v. Lua section that goes into the differences a bit

**Comment by KTom101** [comment]
@yangc95 Added the section.

**Review by yangc95 — APPROVED** [review]
LGTM!

**Review by SSwiniarski — APPROVED** [review]
@KTom101 Looks Good.

**Pull request merged**